lib/optimize/ModuleConcatenationPlugin.js

```diff
@@ -321,29 +321,27 @@ class ModuleConcatenationPlugin {
 	}
 
 	getImports(module) {
-		return Array.from(
-			new Set(
-				module.dependencies
-
-					// Only harmony Dependencies
-					.filter(dep => dep instanceof HarmonyImportDependency)
-
-					// Get reference info for this dependency
-					.map(dep => dep.getReference())
-
-					// Reference is valid and has a module
-					.filter(ref => ref && ref.module)
-
-					// Dependencies are simple enough to concat them
-					.filter(
-						ref =>
-							Array.isArray(ref.importedNames) ||
-							Array.isArray(ref.module.buildMeta.providedExports)
-					)
-
-					// Take the imported module
-					.map(ref => ref.module)
-			)
+		return new Set(
```

Comment:
Isn't it a breaking change since you changed the return type of a plugin API?

Reply:
It's a private method...

Reply:
It would have been obvious in TypeScript...
Is there a roadmap for wp5, I see that mohsen1 is doing a lot of TS stuff...

Reply:
We are a long way away from complete type coverage and producing type definitions from code. But once that's done we can have proper private/public directives to mark what's exposed to the user

Reply:
@mohsen1 "We are a long way away" means wp5 or later :-)?

Reply:
This is the master issue tracking the work:
https://github.com/webpack/webpack/pull/7037

There are two big projects:

1) Typing all of webpack classes using JSDoc 
2) Having TypeScript JSDoc support (Salsa) cover all of the scenarios Webpack might need. For instance supporting `@this` and `@implements` (for plugin classes). 

We've made a lot of progress thanks to @TheLarkInn and amazing TypeScript team but there is a lot more to do. I love to see more people helping with this effort.

Reply:
@mohsen1 please let me know how can I be of any help. Where to track progress, how to split tasks, that kind of things, if they exist? Thanks

Reply:
@mohsen1 If you need help, feel free to assign us to a respective project, sub-ticket we can work on.

Reply:
@mohsen1 after adding jsdoc to a certain file, how to proceed? Open another issue, reference it somewhere? Thanks!

Reply:
TypeScript uses JSDoc as type annotations and will enforce program correctness with them. If you have added types for CachePlugin, make a new PR and ask for review. Running `yarn lint` assures you that your JSDoc is correct. Thanks for contributing!

Reply:
@mohsen1 Here the [PR](https://github.com/webpack/webpack/pull/7518) for the CachePlugin. Feel free to review. Thanks Mohsen!